src/ape/contracts/base.py

```diff
@@ -381,13 +383,54 @@ def __len__(self):
         logs = self.provider.get_contract_logs(self.log_filter)
         return sum(1 for _ in logs)
 
+    def query(
+        self,
+        *columns: List[str],
+        block: int = -1,
+        engine_to_use: Optional[str] = None,
+    ) -> pd.DataFrame:
+        """
+        Iterate through blocks for log events
+
+        Args:
+            block (int): initially set to -1 to grab the last block events
+            engine_to_use (Optional[str]): query engine to use, bypasses query
+              engine selection algorithm.
+
+        Returns:
+            pd.DataFrame
+        """
+
+        stop_block = block
+        if block < 0:
+            stop_block = self.chain_manager.blocks.height + stop_block + 1
+
+        if columns[0] == "*":
+            columns = list(ContractLog.__fields__)  # type: ignore
+
+        start_block = stop_block - 1
+
+        contract_event_query = ContractEventQuery(
+            columns=columns,
+            contract=self.contract.address,
+            event=self.abi,
+            start_block=start_block,
+            stop_block=stop_block,
+        )
+        contract_events = list(
```

Comment:
If you don't convert this to a list, this will be faster when converting to a data frame